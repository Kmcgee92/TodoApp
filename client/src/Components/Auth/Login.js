import React, { useState, useEffect } from "react";
//redux
import { useDispatch, useSelector } from "react-redux";
import { loginHandler } from "../../redux/actions/authActions";
// apollo
import { useMutation } from "@apollo/react-hooks";
// import { GET_USER } from "../../graphql/queries/GetUser";
import { GET_USER_BY_LOGIN } from "../../graphql/mutations/login";

// mui components
import Button from "@material-ui/core/Button";
import Input from "@material-ui/core/Input";
import FormControl from "@material-ui/core/FormControl";
import FormHelperText from "@material-ui/core/FormHelperText";

const Login = ({ classes, setDataLoading, setModalOpen }) => {
  const dispatch = useDispatch();
  const auth = useSelector((state) => state.auth);
  const [email, setEmail] = useState("demo@bcf.com");
  const [password, setPassword] = useState("password");
  const [emailError, setEmailError] = useState(false);
  const [passwordError, setPasswordError] = useState(false);
  const [noUserFound, setNoUserFound] = useState(false);

  let [getUserByLogin, { data, loading, error }] = useMutation(
    GET_USER_BY_LOGIN
  );
  useEffect(() => {
    if (!error && data && !loading) {
      if (!data.Login) {
        setNoUserFound(true);
        return;
      }
      dispatch(loginHandler(data));
    }
  }, [dispatch, error, data, loading]);

  useEffect(() => {
    if (loading) {
      setDataLoading(true);
      //! setting timeout to make loader visible
      // directly based off query for user info loading param
      setTimeout(() => setDataLoading(false), 2800);
    }
  }, [loading, setDataLoading]);

  // error handling for login and perform DB Query which should be done in redux thunk
  const handleLogin =  () => {
    setModalOpen(false);
    setNoUserFound(false);
    setEmailError(false);
    setPasswordError(false);
    let emailCurrentError = false;
    let passwordCurrentError = false;

    if (!email) {
      emailCurrentError = true;
      setEmailError(true);
    }

    if (!password) {
      passwordCurrentError = true;
      setPasswordError(true);
    }

    const regex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

    if (!regex.test(email)) {
      emailCurrentError = true;
      setEmailError(true);
    }

    if (!emailCurrentError && !passwordCurrentError) {
      setDataLoading(true);
      getUserByLogin({
        variables: {
          email: email,
          password: password,
        },
      });
    }
    setDataLoading(false);
  };

  return (
    <div>
      <form className={classes.signinForm}>
        {auth.error && (
          <FormHelperText className={classes.serverError}>
            {`${auth.error}`}
          </FormHelperText>
        )}
        {(emailError || passwordError) && (
          <FormHelperText style={{ color: "red" }}>
            please fill out all fields.
          </FormHelperText>
        )}
        {noUserFound && (
          <FormHelperText style={{ color: "red" }}>
            500: Seems to be an issue with the server
          </FormHelperText>
        )}
        <FormControl>
          <Input
            required
            data-test="loginEmail"
            value={email}
            className={classes.authInputs}
            placeholder="email"
            autoComplete="email"
            autoFocus={true}
            error={emailError}
            onChange={(e) => setEmail(e.target.value)}
            onClick={() => setModalOpen(false)}
            />
        </FormControl>
        <FormControl>
          <Input
            required
            data-test="loginPassword"
            value={password}
            type="password"
            className={classes.authInputs}
            placeholder="password"
            autoComplete="current-password"
            error={passwordError}
            onChange={(e) => setPassword(e.target.value)}
            onClick={() => setModalOpen(false)}
          />
        </FormControl>
        <Button
          data-test="loginButton"
          onClick={handleLogin}
          className={classes.icon}
        >
          Login
        </Button>
      </form>
    </div>
  );
};

export default Login;