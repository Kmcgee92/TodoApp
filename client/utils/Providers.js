import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
// provide redux and apollo
import { Provider } from "react-redux";
import { MockedProvider } from "@apollo/client/testing";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

export default function ComponentWithProviders(
  ComponentToBeTested,
  props,
  initialState
) {
  let mocks = [];
  let store = mockStore(
    
    initialState || { auth: { activeUser: {}, error: false } }
  
  );
  return (
    <MockedProvider mocks={mocks}>
      <Provider store={store}>
        <ComponentToBeTested {...props} />
      </Provider>
    </MockedProvider>
  );
}
