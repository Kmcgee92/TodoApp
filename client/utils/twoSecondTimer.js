export default function twoSecondTmer() {
  return new Promise((resolve) => {
    setTimeout(resolve, 2000);
  });
}
