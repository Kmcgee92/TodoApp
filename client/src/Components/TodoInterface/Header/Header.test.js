import toJSON from "enzyme-to-json";

import { mount } from "enzyme";
// component to test
import Header from "./Header";
// child components
import Login from "../../Auth/Login";
import Logout from "../../Auth/Logout";
import Add from "./Add";
import Delete from "./Delete";

// utils
import Providers from "../../../../utils/Providers";
import { findByTestAtrr } from "../../../../utils/index";

describe("<Header/>", () => {
  let props;
  let wrapWithProviders;
  let component;
  describe("Initial Component Rendering", () => {
    beforeEach(() => {
      props = {
        classes: {},
      };
      wrapWithProviders = mount(Providers(Header, props));
      component = wrapWithProviders.find(Header);
    });
    it("Should Render", () => {
      expect(component).toHaveLength(1);
    });
    it("Should Match Snapshot", () => {
      expect(toJSON(component)).toMatchSnapshot();
    });
  });
  describe("Should Alter Render on Login", () => {
    beforeEach(() => {
      props = {
        classes: {},
      };
      let reduxState = {
        auth: {
          token: "MOCKTOKEN",
          activeUser: { mock: "MOCKUSER" },
        },
      };
      wrapWithProviders = mount(Providers(Header, props, reduxState));
      component = wrapWithProviders.find(Header);
    });
    it("Should Render With Add, Delete, and Logout Components after Login", () => {
      expect(component.find(Logout)).toHaveLength(1);
      expect(component.find(Add)).toHaveLength(1);
      expect(component.find(Delete)).toHaveLength(1);
    });
  });
  describe("Should Render Login", () => {
    beforeEach(() => {
      props = {
        classes: {},
      };
      wrapWithProviders = mount(Providers(Header, props));
      component = wrapWithProviders.find(Header);
    });
    it("Should Render With Login Component on Mount", () => {
      expect(component.find(Login)).toHaveLength(1);
    });
  });
});
