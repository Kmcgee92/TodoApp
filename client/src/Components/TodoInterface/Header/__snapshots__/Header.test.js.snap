// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Initial Component Rendering Should Match Snapshot 1`] = `
<Header
  classes={Object {}}
>
  <div>
    <WithStyles(ForwardRef(AppBar))
      position="fixed"
    >
      <ForwardRef(AppBar)
        classes={
          Object {
            "colorDefault": "MuiAppBar-colorDefault",
            "colorInherit": "MuiAppBar-colorInherit",
            "colorPrimary": "MuiAppBar-colorPrimary",
            "colorSecondary": "MuiAppBar-colorSecondary",
            "colorTransparent": "MuiAppBar-colorTransparent",
            "positionAbsolute": "MuiAppBar-positionAbsolute",
            "positionFixed": "MuiAppBar-positionFixed",
            "positionRelative": "MuiAppBar-positionRelative",
            "positionStatic": "MuiAppBar-positionStatic",
            "positionSticky": "MuiAppBar-positionSticky",
            "root": "MuiAppBar-root",
          }
        }
        position="fixed"
      >
        <WithStyles(ForwardRef(Paper))
          className="MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed"
          component="header"
          elevation={4}
          square={true}
        >
          <ForwardRef(Paper)
            className="MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed"
            classes={
              Object {
                "elevation0": "MuiPaper-elevation0",
                "elevation1": "MuiPaper-elevation1",
                "elevation10": "MuiPaper-elevation10",
                "elevation11": "MuiPaper-elevation11",
                "elevation12": "MuiPaper-elevation12",
                "elevation13": "MuiPaper-elevation13",
                "elevation14": "MuiPaper-elevation14",
                "elevation15": "MuiPaper-elevation15",
                "elevation16": "MuiPaper-elevation16",
                "elevation17": "MuiPaper-elevation17",
                "elevation18": "MuiPaper-elevation18",
                "elevation19": "MuiPaper-elevation19",
                "elevation2": "MuiPaper-elevation2",
                "elevation20": "MuiPaper-elevation20",
                "elevation21": "MuiPaper-elevation21",
                "elevation22": "MuiPaper-elevation22",
                "elevation23": "MuiPaper-elevation23",
                "elevation24": "MuiPaper-elevation24",
                "elevation3": "MuiPaper-elevation3",
                "elevation4": "MuiPaper-elevation4",
                "elevation5": "MuiPaper-elevation5",
                "elevation6": "MuiPaper-elevation6",
                "elevation7": "MuiPaper-elevation7",
                "elevation8": "MuiPaper-elevation8",
                "elevation9": "MuiPaper-elevation9",
                "outlined": "MuiPaper-outlined",
                "root": "MuiPaper-root",
                "rounded": "MuiPaper-rounded",
              }
            }
            component="header"
            elevation={4}
            square={true}
          >
            <header
              className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
            >
              <WithStyles(ForwardRef(Toolbar))>
                <ForwardRef(Toolbar)
                  classes={
                    Object {
                      "dense": "MuiToolbar-dense",
                      "gutters": "MuiToolbar-gutters",
                      "regular": "MuiToolbar-regular",
                      "root": "MuiToolbar-root",
                    }
                  }
                >
                  <div
                    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
                  >
                    <div
                      style={
                        Object {
                          "display": "flex",
                        }
                      }
                    >
                      <WithStyles(ForwardRef(Typography))
                        noWrap={true}
                      >
                        <ForwardRef(Typography)
                          classes={
                            Object {
                              "alignCenter": "MuiTypography-alignCenter",
                              "alignJustify": "MuiTypography-alignJustify",
                              "alignLeft": "MuiTypography-alignLeft",
                              "alignRight": "MuiTypography-alignRight",
                              "body1": "MuiTypography-body1",
                              "body2": "MuiTypography-body2",
                              "button": "MuiTypography-button",
                              "caption": "MuiTypography-caption",
                              "colorError": "MuiTypography-colorError",
                              "colorInherit": "MuiTypography-colorInherit",
                              "colorPrimary": "MuiTypography-colorPrimary",
                              "colorSecondary": "MuiTypography-colorSecondary",
                              "colorTextPrimary": "MuiTypography-colorTextPrimary",
                              "colorTextSecondary": "MuiTypography-colorTextSecondary",
                              "displayBlock": "MuiTypography-displayBlock",
                              "displayInline": "MuiTypography-displayInline",
                              "gutterBottom": "MuiTypography-gutterBottom",
                              "h1": "MuiTypography-h1",
                              "h2": "MuiTypography-h2",
                              "h3": "MuiTypography-h3",
                              "h4": "MuiTypography-h4",
                              "h5": "MuiTypography-h5",
                              "h6": "MuiTypography-h6",
                              "noWrap": "MuiTypography-noWrap",
                              "overline": "MuiTypography-overline",
                              "paragraph": "MuiTypography-paragraph",
                              "root": "MuiTypography-root",
                              "srOnly": "MuiTypography-srOnly",
                              "subtitle1": "MuiTypography-subtitle1",
                              "subtitle2": "MuiTypography-subtitle2",
                            }
                          }
                          noWrap={true}
                        >
                          <p
                            className="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap"
                          >
                            Todo List App
                          </p>
                        </ForwardRef(Typography)>
                      </WithStyles(ForwardRef(Typography))>
                    </div>
                    <div>
                      <Login
                        classes={Object {}}
                      >
                        <div>
                          <form>
                            <WithStyles(ForwardRef(FormControl))>
                              <ForwardRef(FormControl)
                                classes={
                                  Object {
                                    "fullWidth": "MuiFormControl-fullWidth",
                                    "marginDense": "MuiFormControl-marginDense",
                                    "marginNormal": "MuiFormControl-marginNormal",
                                    "root": "MuiFormControl-root",
                                  }
                                }
                              >
                                <div
                                  className="MuiFormControl-root"
                                >
                                  <WithStyles(ForwardRef(Input))
                                    autoComplete="email"
                                    autoFocus={true}
                                    data-test="loginEmail"
                                    error={false}
                                    onChange={[Function]}
                                    onClick={[Function]}
                                    placeholder="email"
                                    required={true}
                                    value="demo@bcf.com"
                                  >
                                    <ForwardRef(Input)
                                      autoComplete="email"
                                      autoFocus={true}
                                      classes={
                                        Object {
                                          "colorSecondary": "MuiInput-colorSecondary",
                                          "disabled": "Mui-disabled",
                                          "error": "Mui-error",
                                          "focused": "Mui-focused",
                                          "formControl": "MuiInput-formControl",
                                          "fullWidth": "MuiInput-fullWidth",
                                          "input": "MuiInput-input",
                                          "inputMarginDense": "MuiInput-inputMarginDense",
                                          "inputMultiline": "MuiInput-inputMultiline",
                                          "inputTypeSearch": "MuiInput-inputTypeSearch",
                                          "marginDense": "MuiInput-marginDense",
                                          "multiline": "MuiInput-multiline",
                                          "root": "MuiInput-root",
                                          "underline": "MuiInput-underline",
                                        }
                                      }
                                      data-test="loginEmail"
                                      error={false}
                                      onChange={[Function]}
                                      onClick={[Function]}
                                      placeholder="email"
                                      required={true}
                                      value="demo@bcf.com"
                                    >
                                      <WithStyles(ForwardRef(InputBase))
                                        autoComplete="email"
                                        autoFocus={true}
                                        classes={
                                          Object {
                                            "colorSecondary": "MuiInput-colorSecondary",
                                            "disabled": "Mui-disabled",
                                            "error": "Mui-error",
                                            "focused": "Mui-focused",
                                            "formControl": "MuiInput-formControl",
                                            "fullWidth": "MuiInput-fullWidth",
                                            "input": "MuiInput-input",
                                            "inputMarginDense": "MuiInput-inputMarginDense",
                                            "inputMultiline": "MuiInput-inputMultiline",
                                            "inputTypeSearch": "MuiInput-inputTypeSearch",
                                            "marginDense": "MuiInput-marginDense",
                                            "multiline": "MuiInput-multiline",
                                            "root": "MuiInput-root MuiInput-underline",
                                            "underline": null,
                                          }
                                        }
                                        data-test="loginEmail"
                                        error={false}
                                        fullWidth={false}
                                        inputComponent="input"
                                        multiline={false}
                                        onChange={[Function]}
                                        onClick={[Function]}
                                        placeholder="email"
                                        required={true}
                                        type="text"
                                        value="demo@bcf.com"
                                      >
                                        <ForwardRef(InputBase)
                                          autoComplete="email"
                                          autoFocus={true}
                                          classes={
                                            Object {
                                              "adornedEnd": "MuiInputBase-adornedEnd",
                                              "adornedStart": "MuiInputBase-adornedStart",
                                              "colorSecondary": "MuiInputBase-colorSecondary MuiInput-colorSecondary",
                                              "disabled": "Mui-disabled Mui-disabled",
                                              "error": "Mui-error Mui-error",
                                              "focused": "Mui-focused Mui-focused",
                                              "formControl": "MuiInputBase-formControl MuiInput-formControl",
                                              "fullWidth": "MuiInputBase-fullWidth MuiInput-fullWidth",
                                              "input": "MuiInputBase-input MuiInput-input",
                                              "inputAdornedEnd": "MuiInputBase-inputAdornedEnd",
                                              "inputAdornedStart": "MuiInputBase-inputAdornedStart",
                                              "inputHiddenLabel": "MuiInputBase-inputHiddenLabel",
                                              "inputMarginDense": "MuiInputBase-inputMarginDense MuiInput-inputMarginDense",
                                              "inputMultiline": "MuiInputBase-inputMultiline MuiInput-inputMultiline",
                                              "inputTypeSearch": "MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch",
                                              "marginDense": "MuiInputBase-marginDense MuiInput-marginDense",
                                              "multiline": "MuiInputBase-multiline MuiInput-multiline",
                                              "root": "MuiInputBase-root MuiInput-root MuiInput-underline",
                                            }
                                          }
                                          data-test="loginEmail"
                                          error={false}
                                          fullWidth={false}
                                          inputComponent="input"
                                          multiline={false}
                                          onChange={[Function]}
                                          onClick={[Function]}
                                          placeholder="email"
                                          required={true}
                                          type="text"
                                          value="demo@bcf.com"
                                        >
                                          <div
                                            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                                            data-test="loginEmail"
                                            onClick={[Function]}
                                            required={true}
                                          >
                                            <input
                                              aria-invalid={false}
                                              autoComplete="email"
                                              autoFocus={true}
                                              className="MuiInputBase-input MuiInput-input"
                                              disabled={false}
                                              onAnimationStart={[Function]}
                                              onBlur={[Function]}
                                              onChange={[Function]}
                                              onFocus={[Function]}
                                              placeholder="email"
                                              required={true}
                                              type="text"
                                              value="demo@bcf.com"
                                            />
                                          </div>
                                        </ForwardRef(InputBase)>
                                      </WithStyles(ForwardRef(InputBase))>
                                    </ForwardRef(Input)>
                                  </WithStyles(ForwardRef(Input))>
                                </div>
                              </ForwardRef(FormControl)>
                            </WithStyles(ForwardRef(FormControl))>
                            <WithStyles(ForwardRef(FormControl))>
                              <ForwardRef(FormControl)
                                classes={
                                  Object {
                                    "fullWidth": "MuiFormControl-fullWidth",
                                    "marginDense": "MuiFormControl-marginDense",
                                    "marginNormal": "MuiFormControl-marginNormal",
                                    "root": "MuiFormControl-root",
                                  }
                                }
                              >
                                <div
                                  className="MuiFormControl-root"
                                >
                                  <WithStyles(ForwardRef(Input))
                                    autoComplete="current-password"
                                    data-test="loginPassword"
                                    error={false}
                                    onChange={[Function]}
                                    onClick={[Function]}
                                    placeholder="password"
                                    required={true}
                                    type="password"
                                    value="password"
                                  >
                                    <ForwardRef(Input)
                                      autoComplete="current-password"
                                      classes={
                                        Object {
                                          "colorSecondary": "MuiInput-colorSecondary",
                                          "disabled": "Mui-disabled",
                                          "error": "Mui-error",
                                          "focused": "Mui-focused",
                                          "formControl": "MuiInput-formControl",
                                          "fullWidth": "MuiInput-fullWidth",
                                          "input": "MuiInput-input",
                                          "inputMarginDense": "MuiInput-inputMarginDense",
                                          "inputMultiline": "MuiInput-inputMultiline",
                                          "inputTypeSearch": "MuiInput-inputTypeSearch",
                                          "marginDense": "MuiInput-marginDense",
                                          "multiline": "MuiInput-multiline",
                                          "root": "MuiInput-root",
                                          "underline": "MuiInput-underline",
                                        }
                                      }
                                      data-test="loginPassword"
                                      error={false}
                                      onChange={[Function]}
                                      onClick={[Function]}
                                      placeholder="password"
                                      required={true}
                                      type="password"
                                      value="password"
                                    >
                                      <WithStyles(ForwardRef(InputBase))
                                        autoComplete="current-password"
                                        classes={
                                          Object {
                                            "colorSecondary": "MuiInput-colorSecondary",
                                            "disabled": "Mui-disabled",
                                            "error": "Mui-error",
                                            "focused": "Mui-focused",
                                            "formControl": "MuiInput-formControl",
                                            "fullWidth": "MuiInput-fullWidth",
                                            "input": "MuiInput-input",
                                            "inputMarginDense": "MuiInput-inputMarginDense",
                                            "inputMultiline": "MuiInput-inputMultiline",
                                            "inputTypeSearch": "MuiInput-inputTypeSearch",
                                            "marginDense": "MuiInput-marginDense",
                                            "multiline": "MuiInput-multiline",
                                            "root": "MuiInput-root MuiInput-underline",
                                            "underline": null,
                                          }
                                        }
                                        data-test="loginPassword"
                                        error={false}
                                        fullWidth={false}
                                        inputComponent="input"
                                        multiline={false}
                                        onChange={[Function]}
                                        onClick={[Function]}
                                        placeholder="password"
                                        required={true}
                                        type="password"
                                        value="password"
                                      >
                                        <ForwardRef(InputBase)
                                          autoComplete="current-password"
                                          classes={
                                            Object {
                                              "adornedEnd": "MuiInputBase-adornedEnd",
                                              "adornedStart": "MuiInputBase-adornedStart",
                                              "colorSecondary": "MuiInputBase-colorSecondary MuiInput-colorSecondary",
                                              "disabled": "Mui-disabled Mui-disabled",
                                              "error": "Mui-error Mui-error",
                                              "focused": "Mui-focused Mui-focused",
                                              "formControl": "MuiInputBase-formControl MuiInput-formControl",
                                              "fullWidth": "MuiInputBase-fullWidth MuiInput-fullWidth",
                                              "input": "MuiInputBase-input MuiInput-input",
                                              "inputAdornedEnd": "MuiInputBase-inputAdornedEnd",
                                              "inputAdornedStart": "MuiInputBase-inputAdornedStart",
                                              "inputHiddenLabel": "MuiInputBase-inputHiddenLabel",
                                              "inputMarginDense": "MuiInputBase-inputMarginDense MuiInput-inputMarginDense",
                                              "inputMultiline": "MuiInputBase-inputMultiline MuiInput-inputMultiline",
                                              "inputTypeSearch": "MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch",
                                              "marginDense": "MuiInputBase-marginDense MuiInput-marginDense",
                                              "multiline": "MuiInputBase-multiline MuiInput-multiline",
                                              "root": "MuiInputBase-root MuiInput-root MuiInput-underline",
                                            }
                                          }
                                          data-test="loginPassword"
                                          error={false}
                                          fullWidth={false}
                                          inputComponent="input"
                                          multiline={false}
                                          onChange={[Function]}
                                          onClick={[Function]}
                                          placeholder="password"
                                          required={true}
                                          type="password"
                                          value="password"
                                        >
                                          <div
                                            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                                            data-test="loginPassword"
                                            onClick={[Function]}
                                            required={true}
                                          >
                                            <input
                                              aria-invalid={false}
                                              autoComplete="current-password"
                                              className="MuiInputBase-input MuiInput-input"
                                              disabled={false}
                                              onAnimationStart={[Function]}
                                              onBlur={[Function]}
                                              onChange={[Function]}
                                              onFocus={[Function]}
                                              placeholder="password"
                                              required={true}
                                              type="password"
                                              value="password"
                                            />
                                          </div>
                                        </ForwardRef(InputBase)>
                                      </WithStyles(ForwardRef(InputBase))>
                                    </ForwardRef(Input)>
                                  </WithStyles(ForwardRef(Input))>
                                </div>
                              </ForwardRef(FormControl)>
                            </WithStyles(ForwardRef(FormControl))>
                            <WithStyles(ForwardRef(Button))
                              data-test="loginButton"
                              onClick={[Function]}
                            >
                              <ForwardRef(Button)
                                classes={
                                  Object {
                                    "colorInherit": "MuiButton-colorInherit",
                                    "contained": "MuiButton-contained",
                                    "containedPrimary": "MuiButton-containedPrimary",
                                    "containedSecondary": "MuiButton-containedSecondary",
                                    "containedSizeLarge": "MuiButton-containedSizeLarge",
                                    "containedSizeSmall": "MuiButton-containedSizeSmall",
                                    "disableElevation": "MuiButton-disableElevation",
                                    "disabled": "Mui-disabled",
                                    "endIcon": "MuiButton-endIcon",
                                    "focusVisible": "Mui-focusVisible",
                                    "fullWidth": "MuiButton-fullWidth",
                                    "iconSizeLarge": "MuiButton-iconSizeLarge",
                                    "iconSizeMedium": "MuiButton-iconSizeMedium",
                                    "iconSizeSmall": "MuiButton-iconSizeSmall",
                                    "label": "MuiButton-label",
                                    "outlined": "MuiButton-outlined",
                                    "outlinedPrimary": "MuiButton-outlinedPrimary",
                                    "outlinedSecondary": "MuiButton-outlinedSecondary",
                                    "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                                    "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                                    "root": "MuiButton-root",
                                    "sizeLarge": "MuiButton-sizeLarge",
                                    "sizeSmall": "MuiButton-sizeSmall",
                                    "startIcon": "MuiButton-startIcon",
                                    "text": "MuiButton-text",
                                    "textPrimary": "MuiButton-textPrimary",
                                    "textSecondary": "MuiButton-textSecondary",
                                    "textSizeLarge": "MuiButton-textSizeLarge",
                                    "textSizeSmall": "MuiButton-textSizeSmall",
                                  }
                                }
                                data-test="loginButton"
                                onClick={[Function]}
                              >
                                <WithStyles(ForwardRef(ButtonBase))
                                  className="MuiButton-root MuiButton-text"
                                  component="button"
                                  data-test="loginButton"
                                  disabled={false}
                                  focusRipple={true}
                                  focusVisibleClassName="Mui-focusVisible"
                                  onClick={[Function]}
                                  type="button"
                                >
                                  <ForwardRef(ButtonBase)
                                    className="MuiButton-root MuiButton-text"
                                    classes={
                                      Object {
                                        "disabled": "Mui-disabled",
                                        "focusVisible": "Mui-focusVisible",
                                        "root": "MuiButtonBase-root",
                                      }
                                    }
                                    component="button"
                                    data-test="loginButton"
                                    disabled={false}
                                    focusRipple={true}
                                    focusVisibleClassName="Mui-focusVisible"
                                    onClick={[Function]}
                                    type="button"
                                  >
                                    <button
                                      className="MuiButtonBase-root MuiButton-root MuiButton-text"
                                      data-test="loginButton"
                                      disabled={false}
                                      onBlur={[Function]}
                                      onClick={[Function]}
                                      onDragLeave={[Function]}
                                      onFocus={[Function]}
                                      onKeyDown={[Function]}
                                      onKeyUp={[Function]}
                                      onMouseDown={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseUp={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchMove={[Function]}
                                      onTouchStart={[Function]}
                                      tabIndex={0}
                                      type="button"
                                    >
                                      <span
                                        className="MuiButton-label"
                                      >
                                        Login
                                      </span>
                                      <WithStyles(memo)
                                        center={false}
                                      >
                                        <ForwardRef(TouchRipple)
                                          center={false}
                                          classes={
                                            Object {
                                              "child": "MuiTouchRipple-child",
                                              "childLeaving": "MuiTouchRipple-childLeaving",
                                              "childPulsate": "MuiTouchRipple-childPulsate",
                                              "ripple": "MuiTouchRipple-ripple",
                                              "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                              "rippleVisible": "MuiTouchRipple-rippleVisible",
                                              "root": "MuiTouchRipple-root",
                                            }
                                          }
                                        >
                                          <span
                                            className="MuiTouchRipple-root"
                                          >
                                            <TransitionGroup
                                              childFactory={[Function]}
                                              component={null}
                                              exit={true}
                                            />
                                          </span>
                                        </ForwardRef(TouchRipple)>
                                      </WithStyles(memo)>
                                    </button>
                                  </ForwardRef(ButtonBase)>
                                </WithStyles(ForwardRef(ButtonBase))>
                              </ForwardRef(Button)>
                            </WithStyles(ForwardRef(Button))>
                          </form>
                        </div>
                      </Login>
                    </div>
                  </div>
                </ForwardRef(Toolbar)>
              </WithStyles(ForwardRef(Toolbar))>
            </header>
          </ForwardRef(Paper)>
        </WithStyles(ForwardRef(Paper))>
      </ForwardRef(AppBar)>
    </WithStyles(ForwardRef(AppBar))>
  </div>
</Header>
`;
